app/assets/javascripts/app/form-validation.js

```diff
@@ -11,6 +11,29 @@ document.addEventListener('DOMContentLoaded', () => {
 
   if (forms.length !== 0) {
     [].forEach.call(forms, function(form) {
+      const buttons = form.querySelectorAll('input[type="submit"]');
```

Comment:
I understand why you want to iterate through the button-set (and still think you shouldn't) but I still think you should scope this to simply [type="submit"] to capture buttons as well. I don't think we can be that sure (or should make it necessary) for the submit to be handled by input rather than button.

Reply:
agreed.